src/System.Private.ServiceModel/src/System/ServiceModel/Dispatcher/StreamFormatter.cs

```diff
@@ -49,7 +49,16 @@ private StreamFormatter(MessageDescription messageDescription, MessagePartDescri
         internal void Serialize(XmlDictionaryWriter writer, object[] parameters, object returnValue)
         {
             Stream streamValue = GetStreamAndWriteStartWrapperIfNecessary(writer, parameters, returnValue);
-            writer.WriteValue(new OperationStreamProvider(streamValue));
+            var streamProvider = new OperationStreamProvider(streamValue);
+            StreamFormatterHelper.WriteValue(writer, streamProvider);
+            WriteEndWrapperIfNecessary(writer);
+        }
+
+        internal async Task SerializeAsync(XmlDictionaryWriter writer, object[] parameters, object returnValue)
+        {
+            Stream streamValue = GetStreamAndWriteStartWrapperIfNecessary(writer, parameters, returnValue);
```

Comment:
Good point. Added GetStreamAndWriteStartWrapperIfNecessaryAsync and WriteEndWrapperIfNecessaryAsync